### DISCUSSION TITLE
Set document not applied inside if

### DISCUSSION THREAD

**Discussion by polarkac** [typst/typst#3760] [General]
### Description

If you put `#set document(...)` inside an if body, documents metadata will not be set in PDF. Is this correct behavior?

I am trying to fix my documents which unintentionally used bug #3676.

Lets say you have 3 documents `story1.typ`, `story2.typ` and `theme.typ`. I want to compile `story1` and `story2` as seperate PDF files, so I put `#set document(...)` on a first line. I also want to use `theme.typ` to compile `story1` and `story2` into one PDF, so i put `#include "story1.typ" and #include "story2.typ"`. This will result in error `document set rules must appear before any content`.

In previous version this would result in success compilation where first `#set document(...)` was used as PDF metadata and others where ignored.

### Reproduction URL

https://typst.app/project/rGxTg3ny8ljV9Cd822tQY3

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by emilyyyylime** [comment]
you can use `set document(...) if condition`, but perhaps there's more to be discussed here regarding how Typst should handle metadata of included source files

**Reply by polarkac** [reply]
Great, this works. Is this `if` variant in docs? How is it different from the `if` with body? This seems like a important difference for some context.

Anyway I agree there should be more discussion about metadata of included files. On the other hand I think my use case is rare. Or maybe there is a proper way to do it? Maybe something like #3723 would help?

**Reply by sijow** [reply]
That's a "set-if" rule as described here: https://typst.app/docs/reference/styling/
